test/rest/message.test.js

```diff
@@ -157,16 +157,14 @@ define(['ably', 'shared_helper', 'async', 'chai'], function (Ably, helper, async
         return originalPublish.apply(channel, arguments);
       };
 
-      Ably.Rest._Http.doUri = function (method, uri, headers, body, params, callback) {
-        originalDoUri(method, uri, headers, body, params, function (err) {
-          if (err) {
-            callback(err);
-            return;
-          }
-          /* Fake a publish error from realtime */
-          callback({ message: 'moo', code: 50300, statusCode: 503 });
-        });
+      Ably.Rest._Http.doUri = async function (method, uri, headers, body, params) {
+        const result = await originalDoUri(method, uri, headers, body, params);
+        if (result.error) {
+          return { error: result.error };
+        }
         Ably.Rest._Http.doUri = originalDoUri;
```

Comment:
Should probably move this line above awaiting for `originalDoUri` result. In the callback version it was executed just after synchronous part of originalDoUri, but before the resulting callback was called. In promise version we now wait for `originalDoUri` to finish.
This probably doesn't affect things much, but it would be closer to the behavior we had in callback version.

Reply:
Good point, will change.